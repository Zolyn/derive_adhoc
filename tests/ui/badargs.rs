
use derive_adhoc::{define_derive_adhoc, derive_adhoc, Adhoc};

#[derive(Adhoc)]
struct DataType {
    foo: u8,
    #[adhoc(hash(skip))]
    bar: Vec<String>,
}

derive_adhoc! {
    DataType:

    fn $unknown() { }

    $(
    )

    ${for fields {
        f ${fname junk}() { }
    }}
}

define_derive_adhoc! {
    Broken =

    type Alias = ${ttype $junk};
}

#[derive(Adhoc)]
#[derive_adhoc(Broken)]
struct ForBroken;

fn main() {
}
