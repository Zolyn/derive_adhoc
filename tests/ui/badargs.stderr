error: unknown derive-adhoc keyword
  --> ui/badargs.rs:13:9
   |
13 |     fn $unknown() { }
   |         ^^^^^^^

error: no contained expansion field determined what to repeat here
  --> ui/badargs.rs:15:6
   |
15 |       $(
   |  ______^
16 | |     )
   | |_____^

error: unexpected arguments to expansion keyword
  --> ui/badargs.rs:19:19
   |
19 |         f ${fname junk}() { }
   |                   ^^^^

error: unexpected arguments to expansion keyword
  --> ui/badargs.rs:22:29
   |
22 |     ${if false { } else { } forbidden}
   |                             ^^^^^^^^^

error: `Xmeta as ...` not allowed in conditions
  --> ui/badargs.rs:24:27
   |
24 |     ${if tmeta(some_path) as lit { }}
   |                           ^^

error: unexpected arguments to expansion keyword
  --> ui/badargs.rs:30:26
   |
30 |     type Alias = ${ttype $junk};
   |                          ^
...
33 | #[derive(Adhoc)]
   |          ----- in this derive macro expansion
   |
   = note: this error originates in the macro `derive_adhoc_template_Broken` which comes from the expansion of the derive macro `Adhoc` (in Nightly builds, run with -Z macro-backtrace for more info)
